src/java/org/apache/cassandra/cql3/restrictions/StatementRestrictions.java

```diff
@@ -645,10 +648,12 @@ else if (!allowFiltering)
          * @param orderings orderings from the select statement
          * @param receiver target restriction builder to receive the additional restrictions
          */
-        private void addOrderingRestrictions(List<Ordering> orderings, RestrictionSet.Builder receiver)
+        private void addOrderingRestrictions(List<Ordering> orderings, RestrictionSet.Builder receiver, boolean allowFiltering)
         {
             List<Ordering> annOrderings = orderings.stream().filter(o -> o.expression.hasNonClusteredOrdering()).collect(Collectors.toList());
 
+            if (annOrderings.size() != 0 && allowFiltering)
```

Comment:
This is overzealous.
The following query will get failed despite no filtering:
```
SELECT * FROM test.tab ORDER BY vec ANN OF [....] ALLOW FILTERING;
```

I understand this is simpler than checking if all the actual expressions can be satisfied by an index:
```
for each expression carried by RowFilter:
  check if indexManager.getBestIndexFor(expression) is present
```

but in that case we should be more precise about the actual limitation (it is not true that one of the predicates requires filtering)